.github/workflows/scan-images.yaml

```diff
@@ -0,0 +1,80 @@
+    name: Scan for vulnerabilities ${{ github.event_name == 'schedule' && '(scheduled)' || '' }}
+    runs-on: ubuntu-latest
+    env:
+      # Scanning RELEASED versions require the project `eraser-dev` as part of the repo name.
+      REGISTRY_POSTFIX: ${{ github.event.inputs.version == '' && 'eraser-test' || 'eraser-dev' }}
+      TAG: ${{ github.event.inputs.version == '' && 'test' || github.event.inputs.version }}
+    timeout-minutes: 15
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@8ca2b8b2ece13480cda6dacd3511b49857a23c09
+        with:
+          egress-policy: audit
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+
+      - name: Build Containers
+        # We only build containers for runs wanting to scan against the running branch.
+        if: github.event.inputs.version == ''
+        run: |
+          echo "Build eraser-manager"
+          make docker-build-manager MANAGER_REPO=${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/eraser-manager MANAGER_TAG=${{ env.TAG }}
+          echo "Build remover"
+          make docker-build-remover REMOVER_REPO=${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/remover REMOVER_TAG=${{ env.TAG }}
+          echo "Build collector"
+          make docker-build-collector COLLECTOR_REPO=${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/collector COLLECTOR_TAG=${{ env.TAG }}
+          echo "Build trivy-scanner"
+          make docker-build-trivy-scanner TRIVY_SCANNER_REPO=${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/eraser-trivy-scanner TRIVY_SCANNER_TAG=${{ env.TAG }}
+
+      - name: Scan remover for vulnerabilities
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/remover:${{ env.TAG }}
+          vuln-type: 'os,library'
+          format: 'table'
+          ignore-unfixed: true
+          exit-code: 1
+      - name: Scan eraser-manager for vulnerabilities
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/eraser-manager:${{ env.TAG }}
+          vuln-type: 'os,library'
+          format: 'table'
+          ignore-unfixed: true
+          exit-code: 1
+      - name: Scan collector for vulnerabilities
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/collector:${{ env.TAG }}
+          vuln-type: 'os,library'
+          format: 'table'
+          ignore-unfixed: true
+          exit-code: 1
+      - name: Scan trivy-scanner for vulnerabilities
+        uses: aquasecurity/trivy-action@0.12.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.REGISTRY_POSTFIX }}/eraser-trivy-scanner:${{ env.TAG }}
+          vuln-type: 'os,library'
+          format: 'table'
+          ignore-unfixed: true
+          exit-code: 1
```

Comment:
if there are vulnerabilities, would it be possible to have the workflow create an issue to update?

Reply:
I couldn't find any examples of creating an issue of the scan results.

I did see a [section of the trivy-action docs](https://github.com/aquasecurity/trivy-action#using-trivy-with-github-code-scanning) saying it's possible to upload the results to **[Github Security](https://github.com/eraser-dev/eraser/security)** using `github/codeql-action`. Not sure if you think we {s/c}hould follow that path 👀. If so, I think I can add it to my other fork and test it out. 

An issue I just thought of is: "Will this workflow  instantly exit on the step where the **first** instance of a Trivy scan fails?". I'm going to see what happens by using my other fork to scan some vulnerable images, to make sure it scans every image even if all are vulnerable 👍🏼

Reply:
The easiest/documented way I found of doing this was to use codeql to upload to GitHub Security tab. I made the updates to do so, although if we want to create an action - or postpone creating an action - and remove this, I can do so 👍🏼 

In the [security tab](https://github.com/not-fabian/eraser/security/code-scanning?query=is%3Aopen+branch%3Amain+sort%3Acreated-desc) we can see all the vulns caught for a scan I did on 1.1.0. Did it on the first release I saw to catch as many as I could.

It's a teeny bit confusing going through them although I didnt' find a way to modify their name, I think it's based on the tricy-generated SERIF. Although going into one of the vulns caught, we can see that the metadata included the run it was part of: https://github.com/not-fabian/eraser/security/code-scanning/37

Reply:
I think the security tab link is private, but from the screenshot it looks good!